### PR TITLE
Bump MSRV to 1.70

### PR DESCRIPTION
**Pull request by djc**
As discussed in #2168.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-proto/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## vNext
 
+- Bump MSRV to 1.70 [#2179](https://github.com/open-telemetry/opentelemetry-rust/pull/2179)
```
Comment (cijothomas) [inline opentelemetry-proto/CHANGELOG.md:5]: please skip this, as proto is already on 1.71.1

File: opentelemetry-zipkin/Cargo.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@ categories = [
 keywords = ["opentelemetry", "zipkin", "tracing", "async"]
 license = "Apache-2.0"
 edition = "2021"
-rust-version = "1.65"
+rust-version = "1.70"
```
Comment (lalitb) [inline opentelemetry-zipkin/Cargo.toml:16]: Nit - Also update the [Readme.md](https://github.com/open-telemetry/opentelemetry-rust/blob/6aad3dbc994b4554d7cc1d667822c9968f5b0b67/opentelemetry-zipkin/README.md?plain=1#L111). Not related to this PR and to be handled separately, we need to be consistent regarding mentioning or not-mentioning msrv in `Readme.md` for all the otel crates.

**Comment by lalitb** [comment]
Thanks for the PR. LGTM with nit comments. There are few places for msrv patching and version pinning, which can be cleaned up now. I will raise a separate PR for that.

**Review by lalitb — APPROVED** [review]

**Comment by cijothomas** [comment]
Merging. The nits can be fixed as part of cleanups required in the patch scipts.

**Pull request merged**